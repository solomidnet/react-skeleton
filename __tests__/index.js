import React, { Component } from "react";
import TestRenderer from "react-test-renderer";
import createSkeleton from "../src";

describe("skeleton", () => {
  let predicate, Skeleton;
  beforeEach(() => {
    predicate = props => {
      return !props.title;
    };

    class Card extends Component {
      render() {
        const { title, description } = this.props;
        return (
          <div>
            <h1>{title}</h1>
            <p>{description}</p>
          </div>
        );
      }
    }

    Skeleton = createSkeleton(() => {
      return {
        title: "__________",
        description: "___________"
      };
    }, predicate)(Card);
  });
  it("should work", () => {
    expect(Skeleton).toBeDefined();
  });
  it("should render dummy data when predicate returns true", () => {
    const wrapper = TestRenderer.create(<Skeleton />);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
  it("should render real data when predicate returns false", () => {
    const wrapper = TestRenderer.create(
      <Skeleton title="Eren Yeager" description="Attack on Titan protagonist" />
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
