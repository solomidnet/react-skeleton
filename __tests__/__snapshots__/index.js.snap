// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`skeleton should render dummy data when predicate returns true 1`] = `
<div>
  <h1>
    __________
  </h1>
  <p>
    ___________
  </p>
</div>
`;

exports[`skeleton should render real data when predicate returns false 1`] = `
<div>
  <h1>
    Eren Yeager
  </h1>
  <p>
    Attack on Titan protagonist
  </p>
</div>
`;
